__tests__/action.test.ts

```diff
@@ -0,0 +1,113 @@
+            function(this, a){
+                startUpTag = this.util.getCurrentTag()
+                a.schedule(0, 0);
+            }
+        );
+
+        this.addReaction(
+            new Triggers(this.a),
+            new Args(this.schedulable(this.f)),
+            function(this, f){
+                let intendedTag: Tag | undefined
+                if (intendedTagDelay === undefined) {
+                    intendedTag = undefined
+                } else {
+                    intendedTag = startUpTag.getLaterTag(intendedTagDelay).getMicroStepsLater(intendedTagMicrostepDelay)
+                }
+                f.schedule(0, 0, intendedTag);
+            }
+        );
+    }
+
+    public setLastTagProvisional(value: boolean) {
+        this._isLastTAGProvisional = value
+    }
+}
+
+describe('Intended tag tests', function () {
+    it('Undefined inteded tag', function() {
+        intendedTagDelay = undefined
+        let app = new ReactorWithFederatePortAction()
+        expect(() => app._start()).toThrowError("FederatedPortAction must have an intended tag from RTI.")
+    })
+    
+    it('Intended tag smaller than current tag', function() {
+        intendedTagDelay = TimeValue.nsec(9)
+        intendedTagMicrostepDelay = 0
+        let app = new ReactorWithFederatePortAction()
+        app.setLastTagProvisional(false)
+        expect(() => app._start()).toThrowError("Intended tag must be greater than current tag. Intended tag")
+    })
+    
+    it('Intended tag equal to current tag', function() {
+        intendedTagDelay = TimeValue.nsec(10)
+        intendedTagMicrostepDelay = 1
+        let app = new ReactorWithFederatePortAction()
+        app.setLastTagProvisional(false)
+        expect(() => app._start()).toThrowError("Intended tag must be greater than current tag. Intended tag: ")
+    })
+    
+    it('Intended tag greater than current tag', function() {
+        intendedTagDelay = TimeValue.nsec(11)
+        intendedTagMicrostepDelay = 0
+        let app = new ReactorWithFederatePortAction()
+        app.setLastTagProvisional(false)
+        app._start()
+    })
+
+    it('Intended tag greater than current tag by microstep', function() {
+        intendedTagDelay = TimeValue.nsec(10)
+        intendedTagMicrostepDelay = 2
+        let app = new ReactorWithFederatePortAction()
+        app.setLastTagProvisional(false)
+        app._start()
+    })
+})
+
+describe('Intended tag tests when TAG is provisional', function () {
+    it('Intended tag smaller than current tag', function() {
+        intendedTagDelay = TimeValue.nsec(9)
+        intendedTagMicrostepDelay = 0
+        let app = new ReactorWithFederatePortAction()
+        app.setLastTagProvisional(true)
+        expect(() => app._start()).toThrowError("Intended tag must be greater than or equal to current tag, when the last TAG is provisional. Intended tag: ")
+    })
+    
+    it('Intended tag equal to current tag', function() {
```

Comment:
The next three tests also have no "expect". What do they test?

Reply:
Ah, these tests are to make sure there's no exception. Could we explicitly expect no exception?

Reply:
I think that would be better. That way it is more explicit what is expected. Adding `.not.toThrow()` should do the trick.

Reply:
Done. Thanks for the suggestion!